### DISCUSSION TITLE
Expansion Direction for Grouped Rows

### DISCUSSION THREAD

**Discussion by brentmclark** [TanStack/table#2510] [General]
## Problem

As of now,  the `expandRows` utility always injects the grouped row into the row collection *above* the expanded rows because that's where it was [told to do so in the utility](https://github.com/tannerlinsley/react-table/blob/master/src/utils.js#L252).  Because of this, the only way to get a grouped row to render below the group of expanded rows is to reorder the data after-the-fact.  It would be ideal if this was a first-class citizen of `react-table` instead.

## Objective

Allow for the grouped row to be rendered below the group instead of above the group.  Like this: 

![Star Wars characters grouped by home world with averaged height and mass](https://p193.p3.n0.cdn.getcloudapp.com/items/7KumWOlD/Image%202020-07-07%20at%208.50.29%20AM.png?v=72313993fdb0b895b388cb3ca95be206)

## Proposal

Create a configuration option that will allow expansion of rows above or below a group.  I have code for all of this already, but I want to follow your process correctly.

**Comment by brentmclark** [comment]
If it helps, I'm happy to put up a PR containing the code I'm holding on to.  It will likely add clarity to this request and be a great place to discuss the proposed solution.

**Reply by gargroh** [reply]
Yeah, this will a good add-on feature, whats the update on this :)

**Reply by brentmclark** [reply]
I got sidetracked by life, so I haven't put up the PR yet.  It's now on my list for hacktoberfest, so... a few weeks!

**Reply by tannerlinsley** [reply]
Good news, v8 will have this natively.

**Reply by brentmclark** [reply]
Thanks for your reply!  I clearly didn't get around to that PR, which seems to be the right call since the functionality will be baked into v8.

Let me know if you want me to share my code for this.  It didn't take much to get working the way I needed.

**Reply by TirthakShahDev** [reply]
Hey @tannerlinsley Can you please provide documentation link where to configure this to show grouped row below the group not above ? Or @brentmclark Can you please share a code link where I can take reference for this ? Thank you :-)

**Reply by shakerii** [reply]
Hey @tannerlinsley How can I use this feature?
